### PR TITLE
circuits: zk-gadgets: fixed-point: Add floor constraint gadget

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds a `constrain_equal_floor` method to the `FixedPointGadget` that constrains a given fixed point to equal a given integral value when floored. This will replace the `constrain_equal_integer_ignore_fraction` that is used throughout the matching engine.

### Testing
- Unit tests pass
- Tested all edge cases for the floor gadget

### CODE REVIEW

File: circuits/src/zk_gadgets/fixed_point.rs [hunk 1]
```diff
@@ -129,6 +147,22 @@ impl MultiproverFixedPointGadget {
...
+        integer: Variable,
+        fabric: &Fabric,
+        cs: &mut MpcPlonkCircuit,
+    ) -> Result<(), CircuitError> {
+        let zero_var = cs.zero();
+        let one = ScalarField::one();
+
+        let lhs_minus_rhs =
+            cs.lc(&[fp.repr, integer, zero_var, zero_var], &[one, -*TWO_TO_M_SCALAR, one, one])?;
+        MultiproverGreaterThanEqZeroGadget::<{DEFAULT_FP_PRECISION}>::constrain_greater_than_eq_zero(lhs_minus_rhs, fabric, cs)
```
Comment (akirillo) [inline circuits/src/zk_gadgets/fixed_point.rs:164]: tiny nit: would it be possible to use `MultiproverBitRangeGadget` here? `constrain_greater_than_eq_zero` just calls out to that, but reading this constraint as "the diff must fit in `DEFAULT_FP_PRECISION` bits" makes a lot more sense than "the diff must be >= 0"

**Review by akirillo — APPROVED** [review]

File: circuits/src/zk_gadgets/fixed_point.rs (hunk 1 above)
Comment (joeykraut) [reply circuits/src/zk_gadgets/fixed_point.rs:164]: yeah good call, forgot this gadget existed
